README.md

```diff
@@ -15,27 +15,23 @@ import (
 )
 
 func main() {
-    f1.Execute()
+    f1.Scenarios().Execute()
 }
 ``` 
 
 This will give you a basic `f1` command line runner with the various running modes (see below) that are bundled with it. You can get more information about the various options available by simply running `go run main.go --help`
 
 ### Writing load tests
-Test scenarios consist of three stages: `Setup`, `Run` and `Teardown`. Setup is called once at the start of a test; this may be useful for generating resources needed for all tests, or subscribing to message queues. Run is called for every iteration of the test, often in parallel with multiple goroutines. Teardown is called once after all iterations complete. These Setup, Run and Teardown functions are defined as types in `f1`:
+Test scenarios consist of two stages: `Setup` and `Run`. Setup is called once at the start of a test; this may be useful for generating resources needed for all tests, or subscribing to message queues. Run is called for every iteration of the test, often in parallel with multiple goroutines. Teardown can be called once after all iterations complete to clean up the setup or/and after each iteration to clean up the run in LIFO order. These Setup and Run functions are defined as types in `f1`:
```

Comment:
Think we need to introduce cleanup rather than talking about teardown. 
```suggestion
Test scenarios consist of two stages: `Setup` and `Run`. Setup is called once at the start of a test; this may be useful for generating resources needed for all tests, or subscribing to message queues. Run is called for every iteration of the test, often in parallel with multiple goroutines. These functions are given a T interface which implements a subset of the standard `Testing.T` library. It can be used for defining assertions, failing a test iteration, context-sensitive logging and to define cleanup functions which will run after the iteration completes, or after the whole test run completes, in the case of cleanup functions defined from the setup function. These Setup and Run functions are defined as types in `f1`:
```